### PR TITLE
[DASH-2293] Add upload dialog

### PR DESCRIPTION
**Pull request by powerczuk**
**Upload files Dialog** 

screen shot 2016-10-25 at 15 31 54

**Upload files button** 
screen shot 2016-10-17 at 13 40 57

### CODE REVIEW

**Comment by powerczuk** [comment]
@wpasinski We need to decide what to do when upload will be finished: Close dialog immediately or do nothing.

**Comment by patrykkopycinski** [comment]
i think, if everything went well, the dialog should be closed automatically

File: src/apps/Hosting/HostingFilesList.js [hunk 1]
```diff
@@ -7,8 +7,9 @@ import { DialogsMixin } from '../../mixins';
 import HostingFilesStore from './HostingFilesStore';
 import HostingFilesActions from './HostingFilesActions';
 
-import { ColumnList, Lists, Dialog, Loading } from '../../common';
+import { ColumnList, Dialog, Lists, Loading } from '../../common';
```
Comment (tomsonpl) [inline src/apps/Hosting/HostingFilesList.js]: 👍

File: src/apps/Hosting/Files.js [hunk 2]
```diff
@@ -155,7 +161,14 @@ const HostingFilesView = React.createClass({
...
             <RaisedButton
-              label="Go to site"
+              label="Upload files"
+              primary={true}
+              icon={<FontIcon className="synicon-cloud-upload" style={{ marginTop: 4 }} />}
+              style={{ marginRight: 10 }}
+              onTouchTap={this.handleShowUploadDialog}
+            />
+            <RaisedButton
+              label="Go to sites"
```
Comment (gh3r) [inline src/apps/Hosting/Files.js]: Please use `label="Go to site"` as it was done previously not `sites`

File: src/apps/Hosting/Files.js [hunk 3]
```diff
@@ -155,7 +161,14 @@ const HostingFilesView = React.createClass({
         >
           <Show if={items.length && !isLoading}>
             <RaisedButton
```
Comment (wpasinski) [inline src/apps/Hosting/Files.js]: put these buttons in a flex div. they are both `inline-block`s and they are not aligned properly due to `<Show>` component. I find that better than adding a vertical align style to every single button inside.

Comment (powerczuk) [inline src/apps/Hosting/Files.js]: I've removed buttons margins and still is ok

**Comment by wpasinski** [comment]
"Send files" button should close the dialog and an upload status should be displayed instead of file list. what do you think?

**Comment by gh3r** [comment]
@wpasinski that's how it looked before, just hide `Upload files`, and `Go To site` buttons while upload is in progress

**Comment by wpasinski** [comment]
@gh3r I mean dialog should display only upload buttons, but the status should stay on a standard page (just like before).

**Comment by powerczuk** [comment]
@wpasinski @gh3r can you check uploading flow?

**Comment by wpasinski** [comment]
@powerczuk For me it would be better if we remove dialog close button (and disable esc key) during an upload. Now it seems like it's broken. Also ripple effect needs to be fixed.

In addition there should be a "Go to site" button visible when there are no files since it will redirect to upload instructions. Even when my hosting is empty I might want to open my site and for example copy an url directly from the browser which is impossible now.

**Comment by gh3r** [comment]
@powerczuk I agree with @wpasinski please apply them, rest looks fine

**Comment by powerczuk** [comment]
@gh3r @wpasinski ok, now we have hidden ```X```,  ```CLOSE``` button and ```ESC key``` during upload.

File: src/apps/Hosting/HostingUploadDialog.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import Reflux from 'reflux';
+
+import Store from './HostingUploadDialogStore';
+
+import HostingFilesEmptyView from './HostingFilesEmptyView';
+
+import { DialogMixin } from '../../mixins';
```
Comment (wpasinski) [inline src/apps/Hosting/HostingUploadDialog.js]: import mixing before stores.

File: src/apps/Hosting/UploadFilesButton.js [hunk 5]
```diff
@@ -94,7 +95,7 @@ class UploadFilesButton extends Component {
               />
             </RaisedButton>
             <RaisedButton
-              label="Choose whole folder from disk*"
+              label="Choose folder *"
```
Comment (wpasinski) [inline src/apps/Hosting/UploadFilesButton.js]: 👍

File: src/apps/Hosting/HostingFilesList.js [hunk 6]
```diff
@@ -238,18 +244,55 @@ const HostingFilesList = React.createClass({
...
+      isLoading,
+      isDeleting,
+      isUploading,
+      lastFileIndex,
+      ...other
+    } = this.props;
+
+    const { _dialogVisible } = this.state;
+
+    if (!items.length || isDeleting || (items.length && errorResponses.length && !_dialogVisible)) {
```
Comment (wpasinski) [inline src/apps/Hosting/HostingFilesList.js]: I think that a separate method for this will be better.

**Comment by powerczuk** [comment]
@gh3r ok, I'll hide the `CLOSE` button, but it's not easy to stop uploading process. User can simply stop uploading by refreshing the site. @wpasinski any suggestions?

**Comment by gh3r** [comment]
@powerczuk @wpasinski what with this branch?

**Comment by powerczuk** [comment]
screen shot 2016-11-04 at 13 38 56

**Comment by powerczuk** [comment]
I can't handle with one thing. When user cancels uploading process, then he wants to do that again, there's a problem with uploading. I was trying to clear app state from previous data but It doesn't work. Do you have any ideas?

<br>
<br>

**Buttons:**
screen shot 2016-11-07 at 13 53 27

**Progress bar with button Cancel/Close**
screen shot 2016-11-07 at 17 47 36

**Pull request merged**